go/cbor.go

```diff
@@ -475,14 +472,33 @@ func (dec *Decoder) decodeText(rv DecodeValue, cborInfo byte, aux uint64) error
 			}
 		}
 	} else {
-		raw := make([]byte, aux)
-		_, err = io.ReadFull(dec.rin, raw)
+		raw, err := dec.readBytes(aux)
+		if err != nil {
+			return err
+		}
 		xs := string(raw)
 		return rv.SetString(xs)
 	}
 	return errors.New("internal error in decodeText, shouldn't get here")
 }
 
+func (dec *Decoder) readBytes(n uint64) ([]byte, error) {
```

Comment:
I actually think that trying to read more than our limit of bytes should error. The caller expects to receive that many bytes, so this change may break the contract and cause panics

Reply:
reframed: this function should always return either n bytes, or error

Reply:
If n is > than what can be read from decoder this will return `ErrUnexpectedEOF` from `io.ReadFull`